FrontEnd/src/components/profileList/ProfileCard.jsx

```diff
@@ -0,0 +1,145 @@
+import css from "./ProfileCard.module.css";
+import { Badge, Typography } from "antd";
+import { useState } from "react";
+import { StarOutlined, StarFilled } from "@ant-design/icons";
+import { PropTypes } from "prop-types";
+import axios from "axios";
+import { useSWRConfig } from "swr";
+const { Paragraph } = Typography;
+
+// FIXME: will be changed once serializer will be changed to return display_name
+const regions = [
+  {
+    key: "Lviv region",
+    value: "Львівська область",
+  },
+  {
+    key: "Kyiv region",
+    value: "Київська область",
+  },
+];
+
+export default function ProfileCard(props) {
+  const isAuthorized = props.isAuthorized;
+  const data = props.data;
+
+  const [isSaved, setIsSaved] = useState(data.is_saved);
+
+  const profile = {
+    id: data.id,
+    companyName: data.name,
+    activities: !data.activities.length ? null : data.activities,
+    region: regions.find((region) => region.key == data.region).value,
+    address: data.address,
+    categories: !data.categories.length ? null : data.categories,
+    isSaved: isSaved,
+    commonInfo: data.common_info,
+  };
+
+  const addressLine = `${profile.region ? profile.region : ""}`;
+  const activitiesLine = profile.activities.map((activity) => activity.name).join(", ");
+
+  const filledStar = (
+    <StarFilled
+      style={{ color: "#FFD800", fontSize: "24px" }}
+      onClick={onStarClick}
+    />
+  );
+  const outlinedStar = (
+    <StarOutlined
+      style={{ color: "#FFD800", fontSize: "24px" }}
+      onClick={onStarClick}
+    />
+  );
+
+  const CategoryBadges = ({ categories }) => {
+    return (
+      <>
+        {categories
+          ? categories.map((category) => (
+              <Badge
+                key={category.id}
+                size="medium"
+                count={category.name.toUpperCase()}
+                style={{
+                  backgroundColor: "#1F9A7C",
+                  fontWeight: 600,
+                  fontFamily: "Inter",
+                  fontSize: 10,
+                }}
+              />
+            ))
+          : ""}
+      </>
+    );
+  };
+
+  // FIXME: fix race condition when switching from saved to all w/saved changes
+  function onStarClick() {
+    axios({
```

Comment:
not-blocking: i would recommend to try swr mutattion

Reply:
I guess done